### PR TITLE
54 pad params

### PR DESCRIPTION
**Pull request by joannacknight**
This PR updates the parameters in the config file for Proxy A-Distance (PAD). I've updated a few of the comments in the PAD file. 

Some comments on the parameter selection:
- `balance_test` should always be set to `True` otherwise we are not calculating the PAD (not sure it makes sense for this to be an option in the config)
- The evaluation on the test dataset was better when `balance_train` was also set to `True`
- I have dropped the polynomial kernel as it was very slow to run compared to the linear and rbf kernels
- The Inception + PCA embedding didn't appear to be effective at providing a representation of the datasets that a classifier could distinguish between - so I have not used this
- The Inception + UMAP embedding was effective, and the resulting PAD calculation was quick to calculate
- The classifier could also produce sensible results when using the matrix embedding; however, it would take hours to run my laptop. If we did want to run with the matrix embedding, then it would probably be best to limit the value of C = 0.1
- I started running the metric on the inception embedding only (without UMAP or PCA) this morning, but gave up after it was running for hours - if this is important then I can set it running overnight sometime.

### CODE REVIEW

**Review by lannelin — APPROVED** [review]
lgtm! Thanks!

**Comment by philswatton** [comment]
>     * `balance_test` should always be set to `True` otherwise we are not calculating the PAD (not sure it makes sense for this to be an option in the config)

Is this true re this no longer being the PAD? Fully agreed that balancing the test set is important for the behaviour of the metric, but iirc this wasn't done in the original PAD paper. I'm fully in favour of the addition, but if the original PAD paper doesn't do this we should probably consider doing one in the way they do it (`balance_train` and `balance_test` set to `False`), and one in improved way (`balance_train` and `balance_test` set to `True`).

Or at least doing it in the improved way only and making clear in the report that the improvements are novel for the ms2 project!

**Review by philswatton — APPROVED** [review]
Looks good to me.

Would be interested in responses to my q above - iirc the original PAD work doesn't balance the train and test sets and so the balancing of both is a novel improvement in the ms2 report.

We should either a) run both the original PAD and our improved version or b) only run our improved version. I suspect the former may be more useful from the perspective of checking how important the improvements are in practice, and thus more informative for the purposes of the report.

Otoh, I could also see the argument for not running it - happy to discuss!

**Comment by joannacknight** [comment]
In Ben-David et al. (2006), the PAD is calculated on two datasets with the same number of records (`m'`) - Section 4 of the paper. My understanding is that the samples have to be the same size for the error to be used to approximate the A-distance, though I would probably need to read through it again to be able to argue my case!

Incidentally, one difference between our work and the paper is that they don't appear to split their data into train and test sets, so they appear to train the classifier and use the error from the training dataset to calculate the PAD. 

It wouldn't hurt to include config with the balances both set to False, and then we don't have to use the results? 

I also think I'll rename the configs to make them consistent with your naming convention, i.e `pad_linear_umap` instead of just `pad_linear`. And also wondering now whether we should include the PCA embeddings just for completeness - again if we are running both for mmd and otdd. 

Thoughts?!

**Comment by philswatton** [comment]
Ahh okay - I defer to you then! The code in the PAD repo doesn't seem to balance them, so I wonder if this is something that got missed out down the line?

Nonetheless, let's stick with the implementation you've done for now and we can revisit later if we want to!

**Comment by joannacknight** [comment]
Ah, I'd forgotten about the PAD repo. I think they sampled with probability of 50% between the two datasets for each of the train, val & test sets, so should be balanced but done in a different way.

I've updated the names of the configs to follow your naming convention, and also for completeness have added in the PCA as we are using it for other metrics. I've only used C=0.1 for PCA as it is a bit slower to run (tbh, the C value didn't make much difference in the results of the UMAP configs, it just runs quickly so doesn't hurt running for more values of C).

**Review by philswatton — APPROVED** [review]
LGTM!
